pkg/types/credentials.go

```diff
@@ -50,16 +50,13 @@ func (c *Credentials) MarshalJSON() ([]byte, error) {
 	if toMarshal.Basic == nil || toMarshal.Basic.Username == "" || toMarshal.Basic.Password == "" {
 		toMarshal.Basic = nil
 	}
-	if toMarshal.TLS == nil || toMarshal.TLS.Certificate == "" || toMarshal.TLS.Key == "" {
-		toMarshal.TLS = nil
-	}
 
```

Comment:
why you removed it?

Reply:
maybe like this?
if toMarshal.TLS == nil || (!UseSMCertificate && (toMarshal.TLS.Certificate == "" || toMarshal.TLS.Key == "" )) {
              toMarshal.TLS = nil
}

Reply:
but it is useless,  because we ahve also "omitmpty " on fields

Reply:
it is used by JSON library.